doc/development.rst

```diff
@@ -37,6 +37,164 @@ The source repository (and current branch) of each of these packages
+branch (usually the master branch). Note that pull requests are a GitHub
+feature and live outside the main git functionality; the `git` program itself
+has no knowledge of them.
+
+Forking the repository
+^^^^^^^^^^^^^^^^^^^^^^
+
+In most cases, it will be necessary to fork the repository (that is, create a
+copy of the repository in your own GitHub account), since creating a pull
+requests requires creating a branch in the repo, which requires write access.
+You can fork a repository directly on the GitHub website.
+
+Creating a new pull request
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Create a new branch for your pull request
+
+   .. code:: bash
+
+      $ cd mirgecom           # or loopy, meshmode, ...
+      $ git branch featureX   # Create new branch for your feature
+      $ git checkout featureX # Activate the branch
+      # Edit files
+
+2. Check that your changes pass tests
+
+   .. code:: bash
+
+      # Install these with 'pip install wheel flake8 pep8-naming flake8-quotes pytest pylint'
+      $ flake8
+      $ pydocstyle
+      $ pytest
+
+3. Commit your changes to the new branch
+
+   .. code:: bash
+
+      $ git commit
+
+4. Push your changes
+
+   .. code:: bash
+
+      $ git push -u
+      # alternatively:
+      $ git push -u origin featureX
+
+   If you do not have write privileges to this repository, you can push
+   the change to your fork instead:
+
+   .. code:: bash
+
+      $ git push -u <forked_repo> featureX
+
+5. Create pull request:
+
+   https://github.com/illinois-ceesd/mirgecom/pulls
+
+   The `base` branch should be the `master` branch of the repo you want to
+   merge into in most cases. The `compare` branch is the branch with your
+   changes.
+
+   When creating the pull request, select at least one reviewer (someone that
+   has knowledge about the code you are modifying), add yourself as the
+   assignee, and choose appropriate labels (if any). Note that this can
+   usually not be done for a PR from a forked repository.
+
+6. After the pull request has been merged, you can delete the branch
+   (locally and remotely):
+
+   .. code:: bash
+
+      $ git branch -d featureX    # delete branch locally
+      $ git push –delete featureX # delete it remotely; or delete in web interface
+
+Updating a pull request
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Commit to the same local branch and push that branch:
+
+.. code:: bash
+
+   $ git commit
+   $ git push
+
+When changing the history of a branch (e.g., by rebasing the branch, or
+by amending a commit that is already pushed), you might need to
+force-push it back to the repository (i.e, ``git push --force``). Please
+use this sparingly.
+
+Reviewing/CI
+^^^^^^^^^^^^
+
+Each pull requests for mirgecom needs one manual approval by a reviewer and
+needs to pass the Continuous Integration (CI) tests before merging. We use
+GitHub actions as the CI provider to test each pull request. The CI tests are
+triggered automatically when a pull request is created or updated.
```

Comment:
Describe how to navigate the CI output (what is Flake8?), what is supposed to happen if a test fails (they fix it), how to mark PRs as draft, and how to request reviews. (Example of a PR that could have used this: https://github.com/inducer/pyrometheus/pull/2)

Reply:
Done.